packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -0,0 +1,156 @@
+import 'terra-base/lib/baseStyles';
+import styles from './ShowHide.module.scss';
+import Button from './_ShowHideButton';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Content in the body of the component that will be shown or hidden
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text that will be visible to the user while the component is collapsed
+   */
+  preview: PropTypes.string.isRequired,
+  /**
+   * Button text that will be displayed when the component is collapsed
+   */
+  collapsedButtonText: PropTypes.string,
+  /**
+   * Button text that will be displayed when the component is expanded
+   */
+  expandedButtonText: PropTypes.string,
+  /**
+   * Icon displayed next to the button text
+   */
+  icon: PropTypes.element,
+  /**
+   * The intl object to be injected for translations. Provided by the injectIntl function.
+   */
+  intl: intlShape.isRequired,
+  /**
+   * Sets the animation for the component when it is expanded or collapsed
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the initial state of the component to open and expanded
+   */
+  isInitiallyOpen: PropTypes.bool,
+  /**
+   * Callback function triggered when the component is collapsed
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when the component is expanded
+   */
+  onOpen: PropTypes.func,
+};
+
+const defaultProps = {
+  icon: <IconChevronDown />,
+  isAnimated: false,
+  isInitiallyOpen: false,
+};
+
+class ShowHide extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: props.isInitiallyOpen,
+    };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  handleOnClick(e) {
+    e.preventDefault();
+
+    // Fire event from toggle handlers
+    if (!this.state.isOpen && this.props.onOpen) {
+      this.props.onOpen();
+    } else if (this.state.isOpen && this.props.onClose) {
+      this.props.onClose();
+    }
+
+    this.setState(prevState => ({
+      isOpen: !prevState.isOpen,
+    }));
+  }
+
+  render() {
+    const {
+      collapsedButtonText,
+      children,
+      expandedButtonText,
+      icon,
+      intl,
+      isAnimated,
+      isInitiallyOpen,
+      onClose,
+      onOpen,
+      preview,
+      ...customProps
+    } = this.props;
+
+    const collapsedText = (collapsedButtonText === undefined) ? intl.formatMessage({ id: 'Terra.showhide.showmore' }) : collapsedButtonText;
+    const expandedText = (expandedButtonText === undefined) ? intl.formatMessage({ id: 'Terra.showhide.hide' }) : expandedButtonText;
+    const buttonText = !this.state.isOpen ? collapsedText : expandedText;
+
+    const showHideClassName = cx([
+      'button',
+      { 'is-open': this.state.isOpen },
+      customProps.className,
+    ]);
+
+    let previewText = '';
+
+    if (preview) {
+      if (preview.slice(-1) === '.') {
+        previewText = `${preview}..`;
+      } else {
+        previewText = `${preview}...`;
```

Comment:
Can we rely on the consumer to provide ellipses with their preview text instead?

Reply:
@tbiethman looks like the mockups in Zeppelin were updated and the ellipses were removed: https://github.com/cerner/terra-core/pull/1910/commits/ba4503889c2bc08e4b305ae11dea6738987a40e3